tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -304,6 +304,11 @@ def is_pass(self, hashes, screenshots, relation, fuzzy):
         assert relation in ("==", "!=")
         if not fuzzy or fuzzy == ((0,0), (0,0)):
             equal = hashes[0] == hashes[1]
+            # sometimes images can have different hashes, but pixels can be identical.
+            if not equal:
+                self.logger.info("Image hashes didn't match, checking pixel differences")
+                max_per_channel, pixels_different = self.get_differences(screenshots)
```

Comment:
What is the performance impact of this on browser that do guarantee the same pixels result in the same compressed image?

Can you trigger full runs of all browsers to see if this results in any unexpected changes, and then also compare the running time?

Reply:
> What is the performance impact of this on browser that do guarantee the same pixels result in the same compressed image?
> 
That codeblock only triggers when the hashes are different, so it will only trigger for reftests that are failing on those browsers. I have tested now that it takes around an extra of _70 milliseconds_ to run when it triggers.

> Can you trigger full runs of all browsers to see if this results in any unexpected changes, and then also compare the running time?

Ok. I can do that.

Reply:
Alright, 70 milliseconds extra per test probably doesn't add up to much in total, but let's see the numbers. If it is measurably slower for some config, we might want to limit this fallback to only browsers where we know it's needed.